File: src/main/java/fr/marstech/mtlinkspray/entity/StandardEntity.java
package fr.marstech.mtlinkspray.entity;

import lombok.*;
import lombok.experimental.Accessors;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * Abstract class representing a standard entity with common properties.
 * This class is used as a base class for other entities in the application.
 */

@EqualsAndHashCode()
@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@SuperBuilder
public abstract class StandardEntity {

    @NonNull
    @Builder.Default
    private LocalDateTime creationDate = LocalDateTime.now();

    @NonNull
    @Builder.Default
    private Boolean isEnabled = true;

    private String description;
    private Map<String, String> metadata;
}

File: src/main/java/fr/marstech/mtlinkspray/entity/StandardEntity.kt
package fr.marstech.mtlinkspray.entity

import java.time.LocalDateTime

/**
 * Abstract class representing a standard entity with common properties.
 * This class is used as a base class for other entities in the application.
 */
interface StandardEntity {
    val id: String
    val creationDate: LocalDateTime
    val expiresAt: LocalDateTime?
    var isEnabled: Boolean
    var description: String?
    var metadata: MutableMap<String, String>
    var author: HistoryItem
    var historyItems: MutableList<HistoryItem>
}
